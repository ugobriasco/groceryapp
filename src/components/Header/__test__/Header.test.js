import React from 'react';
import renderer from 'react-test-renderer';

import {Header, styles} from '../';
import buildStyles from '../../../config/styles';

//do it if you care it
beforeAll(() => {
	buildStyles();
});

describe('Header', () => {

	it('renders correctly', () => {
	const tree = renderer.create(<Header />)

	});

	it ('styles an object', () => {
		expect(typeof styles).toBe('object');
	});

	describe('onPress prop', () => {
		it('should call a function', () => {
			const rendered = renderer.create(<Header onPress={() => console.log('click')}/>);
			expect(rendered).toMatchSnapshot();	
		});

	});

});



