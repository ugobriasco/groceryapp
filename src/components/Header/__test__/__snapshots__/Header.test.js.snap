// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header onPress prop should call a function 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#18bc9c",
      "flexDirection": "row",
      "left": 0,
      "paddingTop": 20,
      "position": "relative",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginBottom": 0,
        "paddingLeft": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      The GroceryApp
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "flex-end",
        "flex": 0.4,
        "opacity": 1,
        "paddingBottom": 5,
        "paddingHorizontal": 20,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Image
      resizeMode="contain"
      source={1}
      style={
        Object {
          "width": 18,
        }
      }
    />
  </View>
</View>
`;
